File: testdata/bench/go/BenchmarkTest00016.go
package bench

import (
	"database/sql"
	"net/http"
)

// Hardcoded query, no user input
func Handler00016(w http.ResponseWriter, r *http.Request) {
	db, _ := sql.Open("sqlite3", ":memory:")
	db.Query("SELECT * FROM users WHERE active = true")
}

File: testdata/bench/kotlin/BenchmarkTest00016.kt
package bench

import io.ktor.server.application.*
import io.ktor.server.response.*
import java.sql.DriverManager

// Safe: integer-only validated input used in query
suspend fun handler00016(call: ApplicationCall) {
    val sort = call.parameters["sort"]
    val allowed = listOf("name", "date", "id")
    val safeSort = if (sort in allowed) sort else "id"
    val conn = DriverManager.getConnection("jdbc:sqlite::memory:")
    val rs = conn.createStatement().executeQuery("SELECT * FROM items ORDER BY $safeSort")
    call.respondText(rs.toString())
}
